### DISCUSSION TITLE
Use xunit.assert.source but prevent Roslyn analyzer overload

### DISCUSSION THREAD

**Discussion by lonix1** [xunit/xunit#2324] [Question] [answered]
I want to add custom assertions to `Assert` so I'm using the `xunit.assert.source` and partial class approach. I've added that to a specific "helper" project. It works.

The problem is that nuget package contains source, so my builds contain hundreds of Roslyn analyzer warnings, and vscode also goes slightly crazy while coding due to the analysis.

Is there a way to prevent this situation?

**Comment by lonix1** [comment, marked as answer]
No matter, I went in a different direction. The above approach works but is too complicated.

Now I have a `public class AssertExtensions: Xunit.Assert`, and I place `using Assert = Xunit.AssertExtensions;` above every test class. Much simpler.

[See here](https://stackoverflow.com/a/16204198/9971404).

**Reply by bradwilson** [reply]
I'm marking your workaround as the answer, but I would be curious to know exactly what you're running into. If you wouldn't mind opening a new issue with a repro project that would be great! Thanks!
